src/ansys/pre_commit_hooks/add_license_headers.py

```diff
@@ -168,6 +173,91 @@ def set_header_args(parser, year, file_path, copyright, template):
     return args
 
 
+def check_exists(changed_headers, parser, values, proj, missing_headers, i):
+    """
+    Check if the committed file is missing its header.
+
+    Parameters
+    ----------
+    changed_headers: int
+        ``0`` if no headers were added or updated.
+        ``1`` if headers were added or updated.
+    parser: argparse.ArgumentParser
+        Parser containing default license header arguments.
+    values: dictionary
+        Dictionary containing the values of files, copyright,
+        template, license, changed_headers, year, and git_repo.
+    proj: project.Project
+        Project to run `REUSE <https://reuse.software/>`_ on.
+    missing_headers: list
+        Committed files that are missing copyright and/or
+        license information in their headers.
+
+    Returns
+    -------
+    int
+        ``0`` if all files contain headers and are up to date.
+        ``1`` if ``REUSE`` changed all noncompliant files.
+    """
+    files = values["files"]
+    year = values["year"]
+    copyright = values["copyright"]
+    template = values["template"]
+
+    if i < len(files):
+        # If the committed file is in missing_headers
+        if files[i] in missing_headers:
+            changed_headers = 1
+            # Run REUSE on the file
+            args = set_header_args(parser, year, files[i], copyright, template)
+            if not args.ignore_license_check:
+                args.license = [values["license"]]
+            header.run(args, proj)
+
+            # Check if the next file is in missing_headers
+            return check_exists(changed_headers, parser, values, proj, missing_headers, i + 1)
+        else:
+            # Update the header
+            with NamedTemporaryFile(mode="w", delete=False) as tmp:
+                args = set_header_args(parser, year, files[i], copyright, template)
+                header.run(args, proj, tmp)
+                tmp.close()
+
+            # Print header was successfully changed if it was modified
+            diff = values["git_repo"].git.diff(files[i], name_only=True)
+            if diff:
+                changed_headers = 1
+                print(f"Successfully changed header of {files[i]}")
+
+            return check_exists(changed_headers, parser, values, proj, missing_headers, i + 1)
+
+    return changed_headers
+
+
+def get_full_paths(file_list):
+    """
+    Update file paths to be absolute paths with system separators.
+
+    Parameters
+    ----------
+    file_list: list
+        List containing committed files.
+
+    Returns
+    -------
+    list
+        List containing the full paths of committed files.
+    """
+    for i in range(0, len(file_list)):
+        if "win" in sys.platform:
+            split_str = file_list[i].split("/")
+            file_list[i] = os.path.abspath(os.path.join(*split_str))
+        else:
+            file_list[i] = os.path.abspath(file_list[i])
+
+    return file_list
```

Comment:
```suggestion
    full_path_files = []
    for file in file_list:
        if "win" in sys.platform:
            split_str = file.split("/")
            full_path_files.append(os.path.abspath(os.path.join(*split_str)))
        else:
            full_path_files.append(os.path.abspath(file))

    return full_path_files
```

Avoid muting functions...